core/jazz_ui/e2e/page-objects/jazzservices.po.ts

```diff
@@ -266,4 +266,86 @@ export class Jazz {
   getMetricesRequestCount() {
     return element(by.xpath('//div[contains(text(),"10")]'));
   }
+  // Environment Locators
+  getRepository() {
+    return element(by.xpath('//div[contains(@class,"det-value repository-link link PlaceHolder")]'));
+  }
+// bitbucket locators
+  bitbucketLogo() {
+  return element(by.xpath('//span[@class="aui-header-logo-device"]'));
+  }
+  createBranch() {
+    return element(by.xpath('//span[@class="aui-icon icon-create-branch"]'));
+  }
+  drp_BranchType() {
+    return element(by.xpath('//a[@id="branch-type"]'));
+  }
+  select_BranchType() {
+    return element(by.xpath('//a[contains(text(),"Feature")]'));
+  }
+  branchName() {
+    return element(by.xpath('//input[@id="branch-name"]'));
+  }
+  btn_CreateBranch() {
+    return element(by.xpath('//input[@id="create-branch-submit"]'));
+  }
+  bitUsername() {
+    return element(by.xpath('//*[@id="j_username"]'));
+  }
+  bitPassword() {
+    return element(by.xpath('//*[@id="j_password"]'));
+  }
+  bitLogin() {
+    return element(by.xpath('//*[@id="submit"]'));
+  }
+  testBranch() {
+    return element(by.xpath('//div[@class="eachBranch col-md-2 col-sm-5"]//div//div[@class="overview-value"]'));
+  }
+  getTestBranch() {
+    return element(by.xpath('//div[@class="eachBranch col-md-2 col-sm-5"]//div//div[@class="overview-value"]'));
+  }
+  bitbucketGetServiceName() {
+    return element(by.css('li:nth-child(2).aui-nav-selected>a'));
+  }
+  createBranchLabel() {
+    return element(by.xpath('//h2[contains(text(),"Create branch")]'));
+  }
+  testBranchStatus() {
+    return element(by.xpath('//div[@class="eachBranch col-md-2 col-sm-5"]//div//span[@title="deployment completed"]'));
+  }
+  drpBitTestBranch() {
+    return element(by.xpath('//button[@id="repository-layout-revision-selector"]'));
+  }
+  selectBitTestBranch() {
+    return element(by.xpath('//a/span[@title="feature/test"]'));
+  }
+  indexFile() {
+    return element(by.xpath('//a[contains(text(),"index.js")]'));
+  }
+  btnEditIndexFile() {
+    return element(by.xpath('//button[@class="aui-button in-browser-edit-button"]'));
+  }
+  editIndexFile() {
+    return element(by.xpath('//div[@class="CodeMirror-code"]'));
+  }
+  getBranchFileCommit() {
+    return element(by.xpath('//button[@title="Commit your changes"]'));
+  }
+  getDialogCommit() {
+    return element(by.xpath('//button[@class=aui-button aui-button-primary commit-button"]'));
+  }
+  //gitlab locators
+  drpGitBranchType() {
+    return element(by.xpath('//a[@class="btn dropdown-toggle has-tooltip"]//i[@class="fa fa-caret-down"]'));
+  }
+  selectGitBranchType() {
+    return element(by.xpath('//a[text()="New branch"]'));
+  }
+  gitBranchName() {
+    return element(by.xpath('//input[@id="branch_name"]'));
+  }
+  btnGitCreateBranch() {
+    return element(by.xpath('//button[contains(text(),"Create branch")]'));
```

Comment:
Wherever available are we using specific id, class, title etc. instead of contains in the xpath search string?